packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -439,6 +458,13 @@ export function commitMount(
       }
       return;
     }
+    case 'html':
+    case 'head':
+    case 'body': {
+      if (enableFloat) {
+        restoreAllStylesResources();
```

Comment:
This gets called pretty often, right? Does it really need to be called on every mount or is this just needed if we fail hydration?

Reply:
Hmm, I was thinking it wouldn't be called often but I guess you are saying that on every navigation assuming there was a keyed component above these top level host components they would get remounted? I suppose that is often. We do need to call it every time the host for the style resources mounts because that mount implies an unmount of the current host. without HostSingletons we need to manage this re-generation of styles ourselves.

Reply:
If you are concerned about the speed I could make the actual insertion algo faster with more code size. It's not optimized because I was thinking this would be primarily on hydration mismatch but I hadn't considered that client navigation might also this pathway which would slow down each navigation.

Reply:
Removed